projects/plugins/jetpack/modules/shortcodes/gravatar.php

```diff
@@ -93,16 +98,60 @@ function jetpack_gravatar_profile_shortcode( $atts ) {
 		return false;
 	}
 
+	$hashed_email = hash( 'sha256', strtolower( trim( $user->user_email ) ) );
+	$cache_key    = 'grofile-' . $hashed_email;
+	$profile      = get_transient( $cache_key );
+
+	if ( empty( $profile ) ) {
+		$profile_url = sprintf(
+			'https://secure.gravatar.com/%s.json',
+			$hashed_email
+		);
+
+		$expire        = 300;
+		$response      = wp_remote_get(
+			esc_url_raw( $profile_url ),
+			array( 'User-Agent' => 'WordPress.com Gravatar Profile Widget' )
+		);
+		$response_code = wp_remote_retrieve_response_code( $response );
+
+		if ( $response_code === 200 ) {
+			$profile = wp_remote_retrieve_body( $response );
+			$profile = json_decode( $profile, true );
+
+			if ( is_array( $profile ) && ! empty( $profile['entry'] ) && is_array( $profile['entry'] ) ) {
+				// Cache for 15 minutes.
+				$expire  = 900;
+				$profile = $profile['entry'][0];
+			} else {
+				// Something strange happened.  Cache for 5 minutes.
+				$profile = array();
+			}
+		} else {
+			// Cache for 15 minutes.
+			$expire  = 900;
+			$profile = array();
+		}
```

Comment:
I wonder if this could be simplified so whenever we have an error, we cache for 5 minutes? Something like this?

```diff
diff --git a/projects/plugins/jetpack/modules/shortcodes/gravatar.php b/projects/plugins/jetpack/modules/shortcodes/gravatar.php
index b56c90e396..a3949e0885 100644
--- a/projects/plugins/jetpack/modules/shortcodes/gravatar.php
+++ b/projects/plugins/jetpack/modules/shortcodes/gravatar.php
@@ -108,34 +108,32 @@ function jetpack_gravatar_profile_shortcode( $atts ) {
 			$hashed_email
 		);
 
-		$expire        = 300;
 		$response      = wp_remote_get(
 			esc_url_raw( $profile_url ),
 			array( 'User-Agent' => 'WordPress.com Gravatar Profile Widget' )
 		);
 		$response_code = wp_remote_retrieve_response_code( $response );
+		$expire        = 300; // Cache any errors for 5 minutes.
+		$profile       = array();
 
 		if ( $response_code === 200 ) {
-			$profile = wp_remote_retrieve_body( $response );
-			$profile = json_decode( $profile, true );
-
-			if ( is_array( $profile ) && ! empty( $profile['entry'] ) && is_array( $profile['entry'] ) ) {
-				// Cache for 15 minutes.
+			$profile = json_decode( wp_remote_retrieve_body( $response ), true );
+
+			if (
+				is_array( $profile )
+				&& ! empty( $profile['entry'] )
+				&& is_array( $profile['entry'] )
+			) {
+				// Sucessfully fetched the profile. Cache for 15 minutes.
 				$expire  = 900;
 				$profile = $profile['entry'][0];
-			} else {
-				// Something strange happened.  Cache for 5 minutes.
-				$profile = array();
 			}
-		} else {
-			// Cache for 15 minutes.
-			$expire  = 900;
-			$profile = array();
 		}
 
 		set_transient( $cache_key, $profile, $expire );
 	}
 
+	// Fetching the profile returned an error. Bail.
 	if ( empty( $profile ) ) {
 		return false;
 	}
```

Reply:
Sure thing, updated.